virtual_ecosystem/models/animal/animal_communities.py

```diff
@@ -1,32 +1,18 @@
-"""The ''animal'' module provides animal module functionality.
-
-Notes:
-- assume each grid = 1 km2
-- assume each tick = 1 day (28800s)
-- damuth ~ 4.23*mass**(-3/4) indiv / km2
-"""
+"""The ''animal'' module provides animal module functionality."""
 
 from __future__ import annotations
 
-import random
 from collections.abc import Callable, Iterable
 from itertools import chain
-from math import ceil
-
-from numpy import timedelta64
 
 from virtual_ecosystem.core.data import Data
-from virtual_ecosystem.core.logger import LOGGER
 from virtual_ecosystem.models.animal.animal_cohorts import AnimalCohort
-from virtual_ecosystem.models.animal.animal_traits import DevelopmentType
 from virtual_ecosystem.models.animal.constants import AnimalConsts
 from virtual_ecosystem.models.animal.decay import CarcassPool, ExcrementPool
 from virtual_ecosystem.models.animal.functional_group import (
     FunctionalGroup,
-    get_functional_group_by_name,
 )
 from virtual_ecosystem.models.animal.plant_resources import PlantResources
-from virtual_ecosystem.models.animal.scaling_functions import damuths_law
 
 
 class AnimalCommunity:
```

Comment:
I don't fully understand why this class is required. The carcass, excrement and plant resources _feel_ like they are redundant, given the per_cell list within the Animal Model? And I don't understand why we want to group by the centroid?